### ISSUE TITLE
Milestone notification confetti effect not showing/delayed.

### ISSUE THREAD

**Issue by develric** [wordpress-mobile/WordPress-Android#19573]
Documenting something noticed while reviewing #19549 PR (but that reproduces for me far before that, happy to get confirmation if this happens to others as well).

### Expected behavior
When receiving a milestone notification for number of views achievement, I would expect the confetti effect to be triggered when opening the notification.

### Actual behavior
When receiving a milestone notification for number of views achievement, the confetti effect is triggered sort of late for me when swiping away from the milestone, instead of when opening the notification detail first time.

(see video below, I went back to 19.0 still with the same behavior, using Pixel 3 Emu on Android 12, but same on S23@Android14)

https://github.com/wordpress-mobile/WordPress-Android/assets/47797566/a8291433-e5f1-47e4-a26f-a80672e963a4

### Steps to reproduce the behavior
- If you do not have a views milestone at hand on your accounts, you can send one yourself for example by following the instructions in this diff `D57226-code` (requires a sandbox)
- Go to Notifications and open the milestone
- Confirm the confetti effect is not triggered and you get a similar behavior on swiping as in the video.

**Comment by develric** [comment]
hey @tiagomar 👋 , would be good to confirm if the issue is reproduceable for others, but I've tentatively placed this issue in the Maintenance week board, assigned a medium priority not because it's breaking a flow per se but thinking of the engagement potential of the confetti effect on celebrating a user achievement. Feel free to adjust things though 🙇

**Comment by tiagomar** [comment]
Hey, @develric! It also feels broken to me. When I first opened the notification I couldn't see the confetti, only after swiping through several notification I could notice the confetti during the transition animation. The issue feels quite low priority but I agree with your take around engagement and it should be an easy fix.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by antonis** [comment]
I confirmed that the issue is still reproducible in the latest (`24.4`) version of the app.
